discussions/prog_repo.md

```diff
@@ -0,0 +1,197 @@
+# L3AF Kernel Function Marketplace
+
+The concept of a L3AF Kernel Function Marketplace is to create a location where
+Kernel Functions from any trusted party can be uploaded and made available for
+others to download.
+
+In the context of L3AF, we define a Kernel Function a kernel eBPF program with
+an optional, cooperative userspace program.
+
+There are many things to consider. At the highest level, the purpose of this
+discussion is to arrive an conclusions for:
+
+- What should we name it
+- Is the Kernel Function Marketplace part of the L3AF project?
+- What should an initial version look like
+- What should a more mature version look like
+
+# What should we name it
+
+"L3AF Kernel Function Marketplace" has been the name used up to this point and
```

Comment:
The name "Kernel Function Marketplace" seems generally disliked. Please discuss alternatives here.

Reply:
Per above (which I probably should delete in preference to this comment but meh), in the 2022-01-05 call the general consensus was that `package manager` was a better name than `marketplace`. It certainly has my vote.

Reply:
When I hear the term `package manager` I'm thinking of things like APT, yum, zypper, etc.  I.e., I think a `package manager` is generally thought of a tool for installing, update, and removing packages.

I think I'd prefer the term `eBPF Package Repository`.

Then, perhaps L3AFD could be considered an eBPF package manager.

Reply:
Agree that "manager" is not the right word for the repository, it's the word for an agent that runs on the node and can use a repository.  +1 to "eBPF Package Repository".